### ISSUE TITLE
Weather data not updating

### ISSUE THREAD

**Issue by msbc42** [bremor/bureau_of_meteorology#62]
HA 2020.12.2 - BOM integration installed via HACS.
Weather Forecast card configured in Lovelace frontend. Weather stats update after HA restart and then no updates, unless integration is reloaded.

**Comment by ZEi20T** [comment]
ive been having this problem for months. i think it only updates once per day for me

**Comment by DavidFW1960** [comment]
What problem exactly? Which entity doesn't update for you? I can see no problems at all with this component..

**Comment by msbc42** [comment]
The Temperature, Humidity and Wind Speed are not updating.
![image](https://user-images.githubusercontent.com/37161584/103499854-a4675700-4e9d-11eb-86ab-ad9ceb7f93c1.png)

**Comment by ZEi20T** [comment]
hmm for me the card isnt updating, but the payload contains the updated timestamps.

sorry, i may need to look into this further and log whats happening

**Comment by bremor** [comment]
> ive been having this problem for months. i think it only updates once per day for me

Version 0.10 should've fixed this. If not please try remove/reinstall integration.

**Comment by bremor** [comment]
> HA 2020.12.2 - BOM integration installed via HACS.
> Weather Forecast card configured in Lovelace frontend. Weather stats update after HA restart and then no updates, unless integration is reloaded.

I don't normally use the weather entity but I will add it to my front end to see if I can replicate the problem

**Comment by ZEi20T** [comment]
> 
> 
> > HA 2020.12.2 - BOM integration installed via HACS.
> > Weather Forecast card configured in Lovelace frontend. Weather stats update after HA restart and then no updates, unless integration is reloaded.
> 
> I don't normally use the weather entity but I will add it to my front end to see if I can replicate the problem

thank you! your data source is way more reliable than others available.

**Comment by bremor** [comment]
Sorry to say, but it's working for me. Can you paste a dump from .storage/core.config_entries, it will look something like this:

```
            {
                "entry_id": "f72f2a4bfeca2fb6149dd9e5d5731138",
                "version": 1,
                "domain": "bureau_of_meteorology",
                "title": "Newport",
                "data": {
                    "latitude": -37.XXXXX,
                    "longitude": 144.XXXXX,
                    "observations_create": true,
                    "observations_basename": "Newport",
                    "observations_monitored": [
                        "temp_feels_like",
                        "rain_since_9am",
                        "humidity"
                    ],
                    "forecasts_create": true,
                    "forecasts_basename": "Newport",
                    "forecasts_days": 5,
                    "forecasts_monitored": [
                        "temp_max",
                        "temp_min",
                        "extended_text",
                        "icon_descriptor",
                        "short_text",
                        "rain_amount_range",
                        "rain_chance"
                    ]
                },
                "options": {},
                "system_options": {
                    "disable_new_entities": false
                },
                "source": "user",
                "connection_class": "cloud_poll",
                "unique_id": null
            },
```

**Comment by ZEi20T** [comment]
Mine looks quite different..........

            {
                "entry_id": "bcfb5970fd8fbec9ba79850fde455567",
                "version": 1,
                "domain": "bureau_of_meteorology",
                "title": "XXX",
                "data": {
                    "latitude": -XX.XXX,
                    "longitude": XX.XXX
                },
                "options": {},
                "system_options": {
                    "disable_new_entities": false
                },
                "source": "user",
                "connection_class": "cloud_poll",
                "unique_id": null
            },

**Comment by bremor** [comment]
You should be able to fix this if you remove and re-add this instance of the integration
![image](https://user-images.githubusercontent.com/34525505/103500691-f27d5a00-4e9f-11eb-8acc-7adbcc7d7cf4.png)

**Comment by ZEi20T** [comment]
awesome, thank you!

re-installing allowed me to add all the sensors and forecast etc. my core_config reflects that now also

dont know why i didnt try a re-install first. ill see how this goes.

cheers, appreciate the reply!

**Comment by ZEi20T** [comment]
ok so the lovelace doesnt appear to be updating.

polling from node red, i see the weather is updating, but the HA card and entities havent for around an hour now:
![image](https://user-images.githubusercontent.com/67130552/103505286-fa43fb00-4ead-11eb-9ab4-ad18e602acb2.png)

2nd most recent poll via node red:
![image](https://user-images.githubusercontent.com/67130552/103505128-9b7e8180-4ead-11eb-9cd1-c65f397a3d51.png)

latest poll:
![image](https://user-images.githubusercontent.com/67130552/103505169-b81ab980-4ead-11eb-9230-9f5c987f04de.png)

times have updated

is there a setting im missing in the lovelace card?

**Comment by msbc42** [comment]
Re-installing seems to have fixed it. I see Temp, wind etc updating in the Lovelace card.

**Comment by msbc42** [comment]
Back to not updating:

![image](https://user-images.githubusercontent.com/37161584/103589784-4f305180-4f40-11eb-9a69-bcfe705536a2.png)

**Comment by msbc42** [comment]
Just noticed this error in the logs:

```

Logger: homeassistant
Source: custom_components/bureau_of_meteorology/sensor.py:180
First occurred: January 4, 2021, 11:50:41 AM (83 occurrences)
Last logged: 10:26:57 AM
Error doing job: Task exception was never retrieved

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 119, in _handle_refresh_interval
    await self.async_refresh()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 193, in async_refresh
    update_callback()
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 105, in _update_callback
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 296, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 323, in _async_write_ha_state
    attr.update(self.device_state_attributes or {})
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 180, in device_state_attributes
    attr[ATTR_DATE] = self.collector.daily_forecasts_data["data"][self.day]["date"]
IndexError: list index out of range

```

**Comment by bremor** [comment]
Yes.. I think there is a bug somewhere. Can you please paste your bureau_of_meteorology config from the .storage/core.config_entries file? Please feel free to remove your lat/lon data before pasting.

**Comment by msbc42** [comment]
Also noticed this error immediately after re-starting HA:

```


Logger: homeassistant.components.sensor
Source: custom_components/bureau_of_meteorology/sensor.py:180
Integration: Sensor (documentation, issues)
First occurred: 10:39:43 AM (2 occurrences)
Last logged: 10:39:43 AM

    Error adding entities for domain sensor with platform bureau_of_meteorology
    Error while setting up bureau_of_meteorology platform for sensor

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 316, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 530, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 101, in async_added_to_hass
    self._update_callback()
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 105, in _update_callback
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 296, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 323, in _async_write_ha_state
    attr.update(self.device_state_attributes or {})
  File "/config/custom_components/bureau_of_meteorology/sensor.py", line 180, in device_state_attributes
    attr[ATTR_DATE] = self.collector.daily_forecasts_data["data"][self.day]["date"]
IndexError: list index out of range


```

**Comment by msbc42** [comment]
Config data:
```
"entry_id": "b027b4b37f0a45845754f2069f1fd3ca",
"version": 1,
"domain": "bureau_of_meteorology",
"title": "Malvern",
"data": {
    "latitude": LAT,
    "longitude": LONG,
    "observations_create": true,
    "observations_basename": "Malvern",
    "observations_monitored": [
        "temp",
        "temp_feels_like",
        "rain_since_9am",
        "humidity",
        "wind_speed_kilometre",
        "wind_direction"
    ],
    "forecasts_create": true,
    "forecasts_basename": "Malvern",
    "forecasts_monitored": [
        "temp_max",
        "temp_min",
        "extended_text",
        "icon_descriptor",
        "mdi_icon",
        "short_text",
        "rain_chance"
    ],
    "forecasts_days": 7
},
"options": {},
"system_options": {
    "disable_new_entities": false
},
"source": "user",
"connection_class": "cloud_poll",
"unique_id": null

```

**Comment by bremor** [comment]
I think I know what is happening, there is another open issue with the same problem. Leave it with me :)

**Comment by ZEi20T** [comment]
thanks mate!

also its not updating again for me too. sorry :)

**Comment by fulhamhigh** [comment]
similar problem for me too

**Comment by bremor** [comment]
Fixed in 0.10.2
https://github.com/bremor/bureau_of_meteorology/releases/tag/0.10.2
